test/p3_tests.jl

```diff
@@ -772,50 +549,44 @@ function test_p3_melting(FT)
         F_rim = FT(0.8)
         ρ_rim = FT(800)
         dt = FT(1)
-        F_liq = FT(0) # process not dependent on F_liq
+
+        state = P3.get_state(params; F_rim, ρ_r = ρ_rim)
+        dist = P3.get_distribution_parameters(state; L = Lᵢ, N = Nᵢ)
 
         T_cold = FT(273.15 - 0.01)
-        rate =
-            P3.ice_melt(p3, vel, aps, tps, Lᵢ, Nᵢ, T_cold, ρₐ, F_rim, ρ_rim, dt)
 
-        TT.@test rate.dNdt == 0
-        TT.@test rate.dLdt == 0
+        rate = P3.ice_melt(dist, vel, aps, tps, T_cold, ρₐ, dt)
+
+        @test rate.dNdt == 0
+        @test rate.dLdt == 0
 
         T_warm = FT(273.15 + 0.01)
-        rate =
-            P3.ice_melt(p3, vel, aps, tps, Lᵢ, Nᵢ, T_warm, ρₐ, F_rim, ρ_rim, dt)
+        rate = P3.ice_melt(dist, vel, aps, tps, T_warm, ρₐ, dt)
 
-        TT.@test rate.dNdt >= 0
-        TT.@test rate.dLdt >= 0
+        @test rate.dNdt >= 0
+        @test rate.dLdt >= 0
 
-        TT.@test rate.dNdt ≈ 172032.67844162227 rtol = 1e-6
-        TT.@test rate.dLdt ≈ 8.601633922081113e-5 rtol = 1e-6
+        # NOTE: All reference values are output from the code.
+        # A failing test indicates that the code has changed.
+        # But if the changes are intentional, the reference values can be updated.
+        @test rate.dNdt ≈ 171953.27800922014 rtol = 1e-6
+        @test rate.dLdt ≈ 8.597663900461007e-5 rtol = 1e-6
 
         T_vwarm = FT(273.15 + 0.1)
-        rate = P3.ice_melt(
-            p3,
-            vel,
-            aps,
-            tps,
-            Lᵢ,
-            Nᵢ,
-            T_vwarm,
-            ρₐ,
-            F_rim,
-            ρ_rim,
-            dt,
-        )
+        rate = P3.ice_melt(dist, vel, aps, tps, T_vwarm, ρₐ, dt)
 
-        TT.@test rate.dNdt == Nᵢ
-        TT.@test rate.dLdt == Lᵢ
+        @test rate.dNdt == Nᵢ
+        @test rate.dLdt == Lᵢ
     end
 end
 
 for FT in [Float32, Float64]
     @info("Testing " * string(FT))
 
+    # state creation
+    test_p3_state_creation(FT)
```

Comment:
Given all the changes to the internals of P3 scheme, do any of the tests below work for `Float32` also?

Reply:
All do, with some minor adjustments!